pkg/util/cache/lru/lru.go

```diff
@@ -0,0 +1,70 @@
+package lru
```

Comment:
Can we not make use of our existing LRU cache implementation?

Reply:
Do you mean the package name?

Reply:
We already have `pkg/storage/chunk/cache/embeddedcache.go` which is an LRU, plus we have `github.com/hashicorp/golang-lru` imported as well. I just want to make sure we've exhausted all other options before including yet another LRU implementation.

Reply:
Oh, I see. yep, I have not seen the embedded cache, I checked all the places where we use `github.com/hashicorp/golang-lru` and that is why I decided to create a wrapper over hashicorp-lru. 

I believe it makes sense to change my wrapper to use embedded cache under the hood, instead of using hasicorp-lru because it provides reach instrumentation... 
I will check if it's possible to refactor the embedded cache, because we need to have an eviction callback and also we need to store the struct as a value, not  `[]byte` , and if it's possible, then I will switch to it. 

Also, I will try to find a better name for my implementation to not collide with other LRU implementations. 
Does it make sense?

Reply:
OK super 👍